src/bindings/python/src/pyopenvino/graph/model.cpp

```diff
@@ -260,52 +261,68 @@ void regclass_graph_Model(py::module m) {
+                        new_shape.second = item.second.cast<ov::PartialShape>();
+                    } else if (py::isinstance<py::list>(item.second) || py::isinstance<py::tuple>(item.second)) {
+                        new_shape.second = Common::partial_shape_from_list(item.second.cast<py::list>());
+                    }
+                    new_shapes.insert(new_shape);
+                }
+                self.reshape(new_shapes);
+            } else if (py::isinstance<py::list>(partial_shapes)) {
+                self.reshape(Common::partial_shape_from_list(partial_shapes.cast<py::list>()));
+            } else {
+                throw py::type_error(
+                    "Incorrect type to reshape model. The following argument types are supported:\n"
+                    "(self: openvino.runtime.Model, partial_shapes: Union[ov.runtime.PartialShape, dict, list])");
+            }
         },
         py::arg("partial_shapes"),
-        R"(
-                :param partial_shapes: Index of Output.
-                :type partial_shapes: PartialShape
-                :return : void
-             )");
+        R"( Reshape model inputs.
 
-    function.def(
-        "reshape",
-        [](ov::Model& self, const std::map<size_t, ov::PartialShape>& partial_shapes) {
-            self.reshape(partial_shapes);
-        },
-        py::arg("partial_shapes"),
-        R"(
+            The allowed types of keys in the `partial_shapes` dictionary are:
 
-                :param partial_shapes: Index of Output.
-                :type partial_shapes: Dict[int, PartialShape]
-                :return: void
-             )");
+            (1) `int'
+            (2) `str`
+            (3) `openvino.runtime.Output`
 
-    function.def(
-        "reshape",
-        [](ov::Model& self, const std::map<std::string, ov::PartialShape>& partial_shapes) {
-            self.reshape(partial_shapes);
-        },
-        py::arg("partial_shapes"),
-        R"(
-                :param partial_shapes: Index of Output.
-                :type partial_shapes: Dict[string, PartialShape]
-                :return: void
-             )");
+            The allowed types of values in the `partial_shapes` are:
 
-    function.def(
-        "reshape",
-        [](ov::Model& self, const std::map<ov::Output<ov::Node>, ov::PartialShape>& partial_shapes) {
-            self.reshape(partial_shapes);
-        },
-        py::arg("partial_shapes"),
-        R"(
-                :param partial_shapes: Index of Output.
-                :type partial_shapes: Dict[Output, PartialShape]
-                :return: void
-             )");
+            (1) `openvino.runtime.PartialShape`
+            (2) `list`
+            (3) `tuple`
+
+            To define a single dimension in the list use:
+
+            (1) `int`
+            (2) `[min, max]`
+            (3) `(min, max)`
+            (4) `range(min, max)`
+            (5) `openvino.runtime.Dimension`
```

Comment:
`range(min, max)` is not the same as `[min, max]`. Please add description about how each form is really interpreted.